copy/multiple.go

```diff
@@ -140,8 +162,34 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 				UpdateDigest:    updatedManifestDigest,
 				UpdateSize:      int64(len(updatedManifest)),
 				UpdateMediaType: updatedManifestType})
-		default:
-			return nil, fmt.Errorf("copying image: invalid copy operation %d", instance.op)
+		case instanceCopyClone:
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			unparsedInstance := image.UnparsedInstance(c.rawSource, &instanceCopyList[i].sourceDigest)
+			instanceDetails, err := updatedList.Instance(instanceCopyList[i].sourceDigest)
+			if err != nil {
+				return nil, fmt.Errorf("Replicating image %d/%d from manifest list: %w", i+1, len(instanceCopyList), err)
+			}
+			addCompressionAlgorithms := []compressionTypes.Algorithm{}
+			optionsClone := options
+			if instance.compressionVariant.Algorithm != nil {
+				// copy options with new compression format
+				optionsClone.DestinationCtx.CompressionFormat = instance.compressionVariant.Algorithm
+				optionsClone.DestinationCtx.CompressionLevel = &instance.compressionVariant.Level
+				addCompressionAlgorithms = append(addCompressionAlgorithms, *instance.compressionVariant.Algorithm)
+			}
+			addManifest, addManifestType, addManifestDigest, err := c.copySingleImage(ctx, policyContext, optionsClone, unparsedToplevel, unparsedInstance, &instanceCopyList[i].sourceDigest, instance.compressionVariant.Algorithm)
+			if err != nil {
+				return nil, fmt.Errorf("Replicating image %d/%d from manifest list: %w", i+1, len(instanceCopyList), err)
+			}
+			// Record the result of a possible conversion here.
+			instanceEdits = append(instanceEdits, internalManifest.ListEdit{
+				ListOperation:            internalManifest.ListOpAdd,
+				AddDigest:                addManifestDigest,
+				AddSize:                  int64(len(addManifest)),
+				AddMediaType:             addManifestType,
+				AddPlatform:              instanceDetails.Platform,
+				AddCompressionAlgorithms: addCompressionAlgorithms,
```

Comment:
(I assume the values of these fields could be prepared in `prepareInstanceCopies`, it has, or will need to have, that data anyway — so populating fields in `instanceCopy` could make this executing loop simpler, and give us more cheap unit test coverage.)